### PR TITLE
Feat/descriptions

### PR DESCRIPTION
**Pull request by kooriookami**
Please make sure these boxes are checked before submitting your PR, thank you!

#1855

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/descriptions/src/descriptions-cell.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      return h('td', {
+        colSpan: this.span,
+      }, [
+        h('span', {
+          class: ['el-descriptions__label', { 'is-bordered-label': this.border }],
+        }, this.label),
+        h('span', {
+          class: 'el-descriptions__content',
+        }, this.content)])
+    }
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-cell.ts:55]: Can we use `switch` here for better readability?

File: packages/descriptions/src/descriptions-row.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <template v-if="border">
+          <el-descriptions-cell :cell="cell" tag="td" type="label" />
+          <el-descriptions-cell :cell="cell" tag="td" type="content" />
+        </template>
+        <template v-else>
+          <el-descriptions-cell :cell="cell" tag="td" type="both" />
+        </template>
+      </template>
+    </tr>
+  </template>
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-row.vue:26]:
```vue
<tr v-else>
</tr>
```

Comment (JeremyWuuuuu) [reply packages/descriptions/src/descriptions-row.vue:26]: We don't need the extra template which renders redundant `Fragment`

Comment (JeremyWuuuuu) [reply packages/descriptions/src/descriptions-row.vue:26]: Same with the `v-else` branch at line 22.

File: packages/descriptions/src/descriptions-row.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  components: {
+    [DescriptionsCell.name]: DescriptionsCell,
+  },
+  props: {
+    row: {
+      type: Array,
+    },
+  },
+  setup() {
+    const descriptions: IDescriptionsInject = inject('descriptions')
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-row.vue:45]: Make this `description` a injection key constant like `el-form` did, and share it with others.

File: packages/descriptions/src/descriptions-row.vue [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    const direction = computed(() => descriptions.direction)
+    const border = computed(() => descriptions.border)
+
+    return {
+      direction: direction,
+      border: border,
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-row.vue:51]:
```ts
return {
  direction,
  border,
}
```
or 

```ts
return {
  direction: computed(() => descriptions.direction),
  border: computed(() => descriptions.border),
}
```

File: packages/descriptions/src/descriptions-row.vue [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  props: {
+    row: {
+      type: Array,
+    },
+  },
+  setup() {
+    const descriptions: IDescriptionsInject = inject('descriptions')
+    const direction = computed(() => descriptions.direction)
+    const border = computed(() => descriptions.border)
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-row.vue:47]: simply return `descriptions` and use it like `descriptions.direction` `descriptions.border` is better than this unnecessary wrapping.

File: packages/descriptions/src/descriptions-cell.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    },
+  },
+  setup(props) {
+    const descriptions: IDescriptionsInject = inject('descriptions')
+
+    const label = computed(() => props.cell?.children?.label?.() || props.cell?.props?.label)
+    const content = computed(() => props.cell?.children?.default?.())
+    const span = computed(() => props.cell?.props?.span || 1)
+    const border = computed(() => descriptions.border)
+    const direction = computed(() => descriptions.direction)
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-cell.ts:24]: Same here, do not create unnecessary reactive objects.

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
LGTM, let's merge it.

**Pull request merged**